Exiled.Events/Patches/Fixes/SinkholeEffectFix.cs

```diff
@@ -0,0 +1,57 @@
+// -----------------------------------------------------------------------
+// <copyright file="SinkholeEffectFix.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Fixes
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.Events.Patches.Events.Player;
+
+    using HarmonyLib;
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="SinkholeEnvironmentalHazard.OnEnter(ReferenceHub)"/>.
+    /// Adds the better effect logic.
+    /// </summary>
+    /// <seealso cref="StayingOnSinkholeEnvironmentalHazard"/>
+    /// <seealso cref="ExitingSinkholeEnvironmentalHazard"/>
+    [HarmonyPatch(typeof(SinkholeEnvironmentalHazard), nameof(SinkholeEnvironmentalHazard.OnEnter))]
+    internal static class SinkholeEffectFix
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            newInstructions.InsertRange(0, new[]
+            {
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Call, Method(typeof(EnvironmentalHazard), nameof(EnvironmentalHazard.OnEnter))),
+            });
+
+            int offset = -2;
+            int index = newInstructions.FindLastIndex(i => i.Calls(Method(typeof(EnvironmentalHazard), nameof(EnvironmentalHazard.OnEnter)))) + offset;
```

Comment:
Just to be clear, as I read this, you're moving the OnEnter to be called immediately, and deleting the original OnEnter later on. Further, you also are not enabling the SinkHole effect OnEnter anymore. Is this an accurate understanding?

Reply:
You're right about `OnEnter` moving, but i still enable effect, I just set duration to 0, cause thats infinity duration

Reply:
Gotcha, thanks for clarifying